src/services/EdFi.DataManagementService.Api/AspNetCoreFrontend.cs

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Text.Json.Nodes;
+using EdFi.DataManagementService.Api.Core;
+using EdFi.DataManagementService.Api.Core.Middleware;
+using EdFi.DataManagementService.Api.Core.Model;
+
+namespace EdFi.DataManagementService.Api
+{
+    /// <summary>
+    /// A thin static class that converts from ASP.NET Core to the DMS facade.
+    /// </summary>
+    public static class AspNetCoreFrontend
+    {
+        /// <summary>
+        /// Takes an HttpRequest and returns a deserialized request body
+        /// </summary>
+        private static async Task<JsonNode?> ExtractJsonBodyFrom(HttpRequest request)
```

Comment:
https://fakeiteasy.github.io/docs/7.4.0/Recipes/faking-http-client/

Related, but not exactly the same and might not be helpful.